core/o11y/src/lib.rs

```diff
@@ -395,6 +412,10 @@ pub async fn default_subscriber_with_opentelemetry(
     let (writer, writer_guard) = tracing_appender::non_blocking(lined_stderr);
 
     let subscriber = tracing_subscriber::registry();
+    // Installs LogCounter as the top layer.
+    // This layer will see all messages even if some of them will be filtered
+    // out and not printed.
```

Comment:
I guess I’m missing the use-case for this change. I originally thought it was supposed to only count the log messages that weren’t filtered, perhaps to give an overview of potential log spaminess and/or to evaluate possible performance impact, but this is counting all of the events regardless, so it probably isn't it. What is it then? Can you please describe what are you planning to use these metrics for in the issue description?

(Furthermore, a caveat: these subscribers won’t see any of the events that are disabled with compile-time filters.)

Reply:
Interesting. Installing this layer as the innermost or the outermost layer has exactly the same result. Conceptually, I want it to be the innermost layer so that it's triggered depending on the `RUST_LOG` filters.
The use case (added to the description) is to detect that something is wrong with a node based on the rate of ERROR and WARN messages, and diagnosing what is wrong by using the second metric with the location.